src/shared/path-lookup.c

```diff
@@ -587,14 +586,9 @@ int lookup_paths_init(
                 if (!add)
                         return -ENOMEM;
 
-                if (l) {
-                        r = strv_extend_strv(&l, add, false);
-                        if (r < 0)
+                r = strv_extend_strv(&paths, add, true);
+                if (r < 0)
                                 return r;
-                } else {
-                        l = add;
-                        add = NULL;
-                }
         }
```

Comment:
Yeah, I know. The code just seemed nicer this way :) With your changes this is initialized once, so I thought it doesn't matter much. I can file a revert if you want.